src/Orleans/Runtime/GrainReference.cs

```diff
@@ -15,6 +15,12 @@ public class GrainReference : IAddressable, IEquatable<GrainReference>, ISeriali
         private readonly string genericArguments;
         private readonly GuidId observerId;
 
+        /// <summary>
+        /// Invoke method options specific to this grain reference instance
+        /// </summary>
+        [NonSerialized]
+        private readonly InvokeMethodOptions invokeMethodOptions;
```

Comment:
Oh, I understand - because `Cast<T>()` uses the copy constructor. That's fine, let's leave this.